Yafc.UI/Core/WindowMain.cs

```diff
@@ -68,9 +68,69 @@ internal class MainWindowDrawingSurface : DrawingSurface {
 
         public override Window window { get; }
 
+        /// <summary>
+        /// Function <c>PickRenderDriver()</c> picks the best rendering backend available on the platform.
+        /// 
+        /// This seems like something that SDL2 should do on its own, since the point of SDL2 is to abstract across platform render
+        /// APIs, and it sort of does - but SDL2 has been around for a really long time and its defaults reflect that. On Windows,
+        /// it will autoselect DirectX/Direct3D 9 if given half a chance; DX9 was of course the version that shipped in 2002 and
+        /// supported Windows 98. This is despite the fact that SDL2 supports DX12 and DX11 where possible, and in 2024 "where possible"
+        /// is really going to be "everywhere" - it just doesn't seem to default select them.
+        /// 
+        /// Instead, we can specify the render driver to use when building a renderer instance; to figure out which one, you have to
+        /// go iterate through all the render drivers the library supports and pick the right one by string comparing its name.
+        /// </summary>
+        /// <param name="flags">
+        /// The flags you were going to/are about to pass to SDL_CreateRenderer, just to make sure the function doesn't pick something
+        /// incompatible (this is paranoia since the major renderers tend to support everything relevant).
+        /// </param>
+        /// <returns>The index of the selected render driver, including 0 (SDL autoselect) if no known-best driver exists on this machine. 
+        /// This value should be fed to the second argument of SDL_CreateRenderer()</returns>
+        private static int PickRenderDriver(SDL.SDL_RendererFlags flags) {
+            nint numRenderDrivers = SDL.SDL_GetNumRenderDrivers();
+            System.Diagnostics.Debug.WriteLine($"Render drivers available: {numRenderDrivers}");
```

Comment:
I have a suspicion that we'd like to use `Logging` ([link to the example](https://github.com/shpaass/yafc-ce/pull/195/files#diff-4e4b0daa2d2b15de7f41d55bd81cfea06c55eb704f13b8dbc8aa8e384ce91cd5R12)).  
@DaleStan, what do you think?

Reply:
Oh whoops you're totally right, I'd been working on random bisect heads for so long I forgot that was added to master

Reply:
I do specifically think this should be logged because with future bug reports it would be helpful to know what render backend they were using.

Reply:
@sfoster1 To clarify my point, your documentation and logging are aweseome. I have nothing against logging.